### PR TITLE
Ensure Debugger tracer always uses absolute file path

### PR DESCRIPTION
**Pull request by hxiong388**
Fix a bug to make sure the Debugger tracer always uses absolute file path when facilitating breakpoints.

Add Debugger integration tests, which also covers this corner case bug.

### CODE REVIEW

**Comment by hxiong388** [comment]
Also added Trace integration tests.

File: google-cloud-debugger/ext/google/cloud/debugger/debugger_c/tracer.c [hunk 1]
```diff
@@ -139,15 +139,23 @@ static void
...
-
     ID callers_id;
     ID breakpoints_hit_id;
+    VALUE matching_result;
+
+    c_trace_path = rb_sourcefile();
+    // Ensure C_trace_path is absolute path
+    trace_path = rb_str_new_cstr(c_trace_path);
+    trace_path = rb_file_expand_path(trace_path, Qnil);
+    c_trace_path = rb_string_value_cstr(&trace_path);
```
Comment (dazuma) [inline google-cloud-debugger/ext/google/cloud/debugger/debugger_c/tracer.c]: Is this going to leak the original c_trace_path, or is that string someone else's responsibility?

File: google-cloud-debugger/integration/debugger_helper.rb [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+    # Setup project based on available ENV variables
+    def setup
+      @vtk_debugger_client = $vtk_debugger_client
+
+      refute_nil @vtk_debugger_client, "You do not have an active debugger vtk client to run the tests."
+      super
+    end
+
+    ##
+    # Create a test gRPC Snappoing
```
Comment (dazuma) [inline google-cloud-debugger/integration/debugger_helper.rb]: Snappoing?

File: google-cloud-debugger/ext/google/cloud/debugger/debugger_c/tracer.c (hunk 1 above)
Comment (hxiong388) [inline google-cloud-debugger/ext/google/cloud/debugger/debugger_c/tracer.c]: All the involved variables should be Ruby objects, and the memory management is handled by Ruby String implementation.

File: google-cloud-debugger/integration/debugger_helper.rb (hunk 2 above)
Comment (hxiong388) [inline google-cloud-debugger/integration/debugger_helper.rb]: Fixed.
